drivers/dai/intel/ssp/ssp.c

```diff
@@ -1748,12 +1752,16 @@ static void dai_ssp_stop(struct dai_intel_ssp *dp, int direction)
 	/* disable SSP port if no users */
 	if (ssp->state[DAI_DIR_CAPTURE] == DAI_STATE_PRE_RUNNING &&
 	    ssp->state[DAI_DIR_PLAYBACK] == DAI_STATE_PRE_RUNNING) {
+#if CONFIG_INTEL_ADSP_CAVS
 		if (!(ssp->clk_active & SSP_CLK_BCLK_ES_REQ)) {
+#endif /* CONFIG_INTEL_ADSP_CAVS */
```

Comment:
Not loving this syntax, putting the if() and brackets into the conditional means that the indentation level now becomes context-dependent.

How about:

```c
if (!IS_ENABLED(CONFIG_INTEL_ADSP_CAVS) || !(ssp->clk_active & SSP_CLK_BCLK_ES_REQ)) {
    // ...
}
```
Alternatively you could define a `bool clear_rse_bits;` or something to gate the tests, and set that based on preprocessor state.

Reply:
> 

For me, it's more readable to use if/endif statement, but using a macro is also a good idea...

Reply:
> Or this:
> 
> ```
> bool clear_rse_bits = COND_CODE_1(CONFIG_INTEL_ADSP_CAVS, (!(ssp->clk_active & SSP_CLK_BCLK_ES_REQ)), (false));
> ```

Done :)